contracts/p1/Main.sol

```diff
@@ -49,6 +56,33 @@ contract MainP1 is Versioned, Initializable, Auth, ComponentRegistry, UUPSUpgrad
         stRSR.payoutRewards();
     }
 
+    /// Set Version Registry
+    /// @dev Can only be called once.
+    function setVerionRegistry(VersionRegistry versionRegistry_) external onlyRole(OWNER) {
```

Comment:
typo -- setVersionRegistry

Reply:
Remind me it's not a good idea to write code too late into the day.